Packs/XMCyber/Integrations/XMCyberIntegration/XMCyberIntegration.py

```diff
@@ -714,53 +719,67 @@ def is_xm_version_supported_command(xm: XM, args: Dict[str, Any]) -> CommandResu
     )
 
 
+def update_command_results(
+    xm: XM,
+    command_results: List[CommandResults],
+    xm_data_list: List[Dict[str, Any]],
+    readable_output,
+    entity: Dict[str, Any],
+):
+    id_ = entity.get("entityId")
+    try:
+        ip = entity.get("ipv4Str", "")
+        domain = (
+            entity.get("customProperties", {})
+            .get("domainWorkgroup", {})
+            .get("data", "")
+        )
+        os = entity.get("os", {}).get("type", "")
+        os_version = entity.get("os", {}).get("name", "")
+        hostname = entity.get("displayName", "")
+        endpoint_standard_context = Common.Endpoint(
+            id_,
+            ip_address=ip,
+            domain=domain,
+            os=os,
+            os_version=os_version,
+            hostname=hostname,
+        )
+    except (TypeError, AttributeError, KeyError):
+        endpoint_standard_context = Common.Endpoint(id_)
```

Comment:
It's not a new function, just some refactor to avoid code duplication.
I can fail if the api response is in unexpected format. In that case, we still want to get the endpoint, even without extra info